### PR TITLE
new(build): enable ASan and UBSan builds with options and in CI

### LINKED ISSUE #2930: Support sanitizers in Falco tests CI
**Motivation**

Some time ago we improved testing in Falco, not just unit but integration as well. After we have enabled AddressSanitizer in libs more than a year ago we need to do the same in Falco. For instance, issues like this https://github.com/falcosecurity/falco/pull/2877 should be caught before they are merged.

### PR DESCRIPTION
**Pull request by LucaGuerra**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

/kind feature

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

/area CI

**What this PR does / why we need it**:

This PR adds the following:
* `USE_ASAN` and `USE_UBSAN` options to Falco cmake
* Automated unit tests are ran with both ASan and UBSan enabled on all supported architectures in CI
* Integration tests are ran with ASan on x86_64, no sanitizers on ARM64 (falcosecurity/testing fails with no errors on that arch with sanitizers enabled, needs much deeper investigation)
* libs and drivers are bumped to the latest version

Supersedes https://github.com/falcosecurity/falco/pull/3104

**Which issue(s) this PR fixes**:

Fixes #2930

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by LucaGuerra** [comment]
/hold

Let's merge https://github.com/falcosecurity/falco/pull/3151 first, then I'll rebase this and it'll hopefully pass

File: .github/workflows/reusable_build_packages.yaml [hunk 1]
```diff
@@ -67,17 +77,21 @@ jobs:
           curl -L https://github.com/Kitware/CMake/releases/download/v3.22.5/cmake-3.22.5-linux-$(uname -m).tar.gz \
             | tar --directory=/usr --strip-components=1 -xzp
 
+      - name: Check sanitizer value
```
Comment (FedeDP) [inline .github/workflows/reusable_build_packages.yaml:80]: Debug?

Comment (FedeDP) [reply .github/workflows/reusable_build_packages.yaml:80]: (i'd drop it!)

Comment (LucaGuerra) [reply .github/workflows/reusable_build_packages.yaml:80]: thanks, yes that's a leftover

**Review by FedeDP — APPROVED** [review]
/approve

**Review by incertum — APPROVED** [review]
/approve

This is great, thanks for implementing it @LucaGuerra!

**Pull request merged**